### PR TITLE
updates & various optimization

### PR DESCRIPTION
**Pull request by MJx0**
-updated KittyMemory
-fixed multiple instances for switches and menu variables (declaration at Macros.h) 
-fixed import issues between SwitchTemplate and Menu (SwitchTemplate is merged into Menu file)
-fixed compilation for mobile theos

### CODE REVIEW

File: template/Makefile [hunk 1]
```diff
@@ -1,25 +1,58 @@
...
+# only set this  to 1 if you are on mobile theos
+# assuming you have an sdk at your theos sdks directory
+# this will include c++ headers and other needed headers for your project so you don't need to manually include them or something like that
+# if some c++ headers are still missing in your sdk like "initializer_list" then manually copy them to your c++ headers directory and not your project folder
+# for example in my case c++ headers directory is located at /private/var/theos/sdks/iPhoneOS11.2.sdk/usr/include/c++/4.2.1/
+# please note, do not include c++ headers in your theos includes to enable c++ which is a ghetto solution and use this approach instead
+MOBILE_THEOS=1
+ifeq ($(MOBILE_THEOS),1)
+  # path to your sdk
+  SDK_PATH = /private/var/theos/sdks/iPhoneOS11.2.sdk/
```
Comment (joeyjurjens) [inline template/Makefile]: I think it's better to use: `$(THEOS)/sdks/iPhoneOS11.2.sdk/` here, as every installation guide creates this variable

**Comment by joeyjurjens** [comment]
Hi! Thankyou very much for this pull request, it makes things allot cleaner!
I've left one comment in the makefile though, if you agree & could add that in your changes I'll merge that with it :)

File: template/Makefile (hunk 1 above)
Comment (MJx0) [reply template/Makefile]: sure, I replaced it.

Comment (joeyjurjens) [reply template/Makefile]: Awesome, gonna merge it. Thanks again 👍

**Pull request merged**